flocker/ca/_ca.py

```diff
@@ -106,7 +110,7 @@ def create_certificate_authority(keypair, dn, request, serial, expiry, digest):
     cert.set_subject(req.get_subject())
     cert.set_pubkey(req.get_pubkey())
     cert.gmtime_adj_notBefore(0)
-    cert.gmtime_adj_notAfter(expiry)
+    cert.set_notAfter(expiry)
```

Comment:
This should be `cert.set_notBefore(beginning_of_validity_period)` and `cert.set_notAfter(beginning_of_validity_period + validity_interval)`.

`beginning_of_validity_period` should be a parameter accepted by this function.  `validity_interval` can be hard-coded at the 20 year period as defined in the globals.

This makes it very easy to test.  You know exactly what values to expect in these fields because you passed them in (or you passed in the first and the second is a trivial function of the first).  This removes the factor of "what time was it when the tests started to run" from the equation entirely.

It's kind of like how the sys module is parameterized to the script runner.  The actual sys module is complicated and hard to test - so we replace it with something simple and easy to test.